src/count-islands/index.js

```diff
@@ -1,3 +1,49 @@
-function countIslands(grid) {}
+function countIslands(grid) {
+  const island = [];
+  let islandCount = 0;
+  const toCheck = [];
+
+  function checkNeighbours(row, col) {
```

Comment:
move the checkNeighbours() function out of countIsLands() function, a function should have a single responsibility that way it is easily maintainable and reusable. right now the countIsLand() is getting too complex so many things going on in there